src/drivers/intel/cavs/dmic.c

```diff
@@ -628,7 +628,7 @@ static int select_mode(struct dai *dai,
 	/* Calculate remaining gain to FIR in Q format used for gain
 	 * values.
 	 */
-	fir_in_max = (1 << (DMIC_HW_BITS_FIR_INPUT - 1));
+	fir_in_max = 1 << (DMIC_HW_BITS_FIR_INPUT - 1);
```

Comment:
For this one a better idea would be to use the BIT macro, as in `BIT(DMIC_HW_BITS_FIR_INPUT - 1)`. That has the proper parentheses to ensure semantic isn't changed.

Reply:
Yes, I am doing it in a different patch set.

Reply:
Sure

Reply:
Just a reminder to myself. I wonder why I have placed the extra parentheses? I need to check that it shouldn't be e.g.

fir_in_max = (1 << (DMIC_HW_BITS_FIR_INPUT - 1)) - 1;

I'm not so keen with BIT() since this is not a bitfield with unsigned values, the result is used for arithmetic. Some new macro like POW2(x) would fit better the context.

Reply:
But since you are not changing the functionality you can proceed. I can separately check the impacted filter scaler code.

Reply:
I looked the code, the line should be

`fir_in_max = (1 << (DMIC_HW_BITS_FIR_INPUT - 1)) - 1;`

Could this commit be excluded for the pull request since this is a mistake in original code and redoing it without parentheses doesn't improve things. It likely has not triggered issues with any of the used configurations for DMIC but I need to check.

Instead of using BIT, I'd prefer to replace this with macro that provides for any N bits the what stdint.h INT32_MAX and INT32_MIN provide. Maybe INT_MAX(N) or INT_MIN(X)?